File: Frontend/manga-world/src/Model/MangaFile.ts
import Chapter from "./Chapter";

class MangaFile {
  mfId: number;
  mfFile: string;
  chapter: Chapter;

  constructor(mfId: number, mfFile: string, chapter: Chapter) {
    this.mfId = mfId;
    this.mfFile = mfFile;
    this.chapter = chapter;
  }
}

export default MangaFile;

File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/Entity/MangaFile.java
package com.amakakeru.mangaworld.Entity;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "manga_file")
public class MangaFile {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "mf_id")
    private Long mfId;

    @Column(name = "mf_file", nullable = false)
    private String mfFile;

    @OneToOne
    @JoinColumn(name = "c_id", nullable = false)
    private Chapter chapter;
}
